prov/efa/src/rxr/rxr_msg.c

```diff
@@ -328,13 +287,19 @@ ssize_t rxr_msg_generic_send(struct fid_ep *ep, const struct fi_msg *msg,
 	}
 
 	tx_entry = rxr_ep_alloc_tx_entry(rxr_ep, msg, op, tag, flags);
-
 	if (OFI_UNLIKELY(!tx_entry)) {
 		err = -FI_EAGAIN;
 		rxr_ep_progress_internal(rxr_ep);
 		goto out;
 	}
 
+	efa_ep = container_of(rxr_ep->rdm_ep, struct efa_ep, util_ep.ep_fid);
+	ret = efa_ep_use_p2p(efa_ep, tx_entry->desc[0]);
```

Comment:
can you rename `ret` to `use_p2p`, similarly for the one used in `rxr_msg_post_rtm`.

Reply:
Fixed in new revision. though I think we need to discussion whether check `use_p2p` on send path is necessary.

Reply:
Thx, the current implementation looks good to me.